src/window.cpp

```diff
@@ -20,8 +21,17 @@ window::window (const std::vector<dxp_socket_desktop> &desktops)
 {
   this->xcb_id = xcb_generate_id (drawable::c);
   this->desktops = desktops;
+  this->windows = get_windows (window::c, window::root);
   this->pres = 0; ///< id of the preselected desktop
 
+  if (!dxp_do_screenshots)
+    {
+      // TODO(mmskv): Initialize this->desktops
```

Comment:
is it meant to be empty? why review unfinished work?

Reply:
This is still a draft pull request. Of course it will have unfinished work.